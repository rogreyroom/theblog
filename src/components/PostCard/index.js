import React, { useEffect } from 'react'
import { Link } from 'react-router-dom'
import { useSelector, useDispatch } from 'react-redux'
import PropTypes from 'prop-types'
import { photosActions } from '../../redux/photos/photosSlice'
import { postsActions } from '../../redux/posts/postsSlice'
import classes from './index.module.scss'

const PostCard = ({ id, title }) => {
  const { photos, photo } = useSelector((state) => state.photos)
  const dispatch = useDispatch()

  useEffect(() => {
    if (photos.length > 0) {
      dispatch(photosActions.getPhotoById(id))
    }
  }, [dispatch, photos])

  const setPostData = (postId) => {
    dispatch(postsActions.getPostById(postId))
  }

  return (
    <Link
      to={`blog/${id}`}
      onClick={() => setPostData(id)}
      className={classes.card}
    >
      <img
        className={classes.card__image}
        src={photo.thumbnailUrl}
        alt={photo.title}
      />
      <p className={classes.card__title}>{title}</p>
    </Link>
  )
}

PostCard.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
}

export default PostCard
