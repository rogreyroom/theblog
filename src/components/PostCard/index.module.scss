.card {
  --border-color: transparent;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  background-color: var(--c-grey-alpha);
  border: solid 3px var(--border-color);
  padding: var(--xs);
  text-decoration: none;
  transition: border-color linear .3s;
  width: 100%;

  &__image {
    margin: 0 var(--xs) 0 0;
    max-width: clamp(var(--xxl), 1rem + 10vw, calc(var(--xxl) * 2));
    max-height: clamp(var(--xxl), 1rem + 10vw, calc(var(--xxl) * 2));
    width: 100%;
  }

  &__title {
    color: var(--c-black);
    font-size: clamp(var(--fs-text-small), 0.7212rem + 0.7692vw,  var(--fs-text-large));
    margin: 0;
  }

  &:hover {
    --border-color: var(--c-accent);
  }
}