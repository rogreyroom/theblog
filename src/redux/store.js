import { configureStore } from '@reduxjs/toolkit'
import postsSlice from './posts/postsSlice'
import commentsSlice from './comments/commentsSlice'

const store = configureStore({
  reducer: {
    posts: postsSlice.reducer,
    comments: commentsSlice.reducer,
  },
})

export default store
