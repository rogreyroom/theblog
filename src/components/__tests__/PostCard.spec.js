/**
 * @jest-environment jsdom
 */
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { screen, render, cleanup } from '@testing-library/react'
import { Provider } from 'react-redux'
import PostCard from '../PostCard'
import store from '../../redux/store'

describe('PostCard component', () => {
  let element
  beforeEach(() => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <PostCard id={1} title="Some cool title" />
        </Router>
      </Provider>,
    )
    element = container
  })

  it('renders the component', () => {
    expect(element).toMatchSnapshot()
  })

  it('component contains text "Some cool title"', () => {
    expect(screen.getByText(/title/i)).toHaveTextContent('Some cool title')
  })

  afterAll(cleanup)
})
