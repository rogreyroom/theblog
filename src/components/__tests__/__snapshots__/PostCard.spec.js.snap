// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard component renders the component 1`] = `
<div>
  <a
    class="card"
    href="/blog/1"
  >
    <img
      class="card__image"
    />
    <p
      class="card__title"
    >
      Some cool title
    </p>
  </a>
</div>
`;
